up-core-api/uprotocol/core/udiscovery/v3/udiscovery.proto

```diff
@@ -14,340 +14,128 @@
+  // get ServiceTopic for all topics and it is not permitted to do so
+  rpc GetServiceTopics(GetServiceTopicsRequest) returns (GetServiceTopicsResponse) {
+    option (uprotocol.method_id) = 2;
   }
-}
-
 
-// Typedef for a node properties. A node property can be any one of the uProtocol ("u_") types
-// defined below
-message PropertyValue {
-  oneof attr {
-    bool u_boolean = 1;       // Boolean
-    int32 u_integer = 2;      // Integer
-    string u_string = 3;      // String
-    bytes u_bytes = 4;        // Raw Bytes
-    string u_uri = 5;         // A URI
-    google.protobuf.Timestamp u_timestamp = 6;  // Timestamp
-    double u_double = 7;      // Double
-    int64 u_integer_64 = 8;   // 64 bit Integer
-  }
 }
 
 
-// Node can be domain, device, service, resource, method, etc...
-message Node {
-  // uProtocol long form URI pointing to this node
-  string uri = 1;
-
-  // List of child nodes under this node
-  repeated Node nodes = 2;
+// Request to get information about a topic
+// Example might be to get information about a specific topic, all topics for a service (using wildcard resource_id),
+// or all topics for all services for a given device (using wildcard ue_id and resource_id)
+message GetServiceTopicsRequest {
+  // The URI of the topics that we would like to query. for example, 
+  // to fetch a specific topic, we would set 
+  uprotocol.v1.UUri topic = 1;
 
-  // List of node properties
-  map <string, PropertyValue>  properties = 3;
-
-  // The node type
-  Type type = 4;
-
-
-  // What is the uThing (stored in Node) type. This is used to more easily
-  // identify the Node rather than parsing from uri and inferring the type
-  enum Type {
-    UNSPECIFIED  = 0; // Unspecified node type
-    DOMAIN = 1;       // uDomain
-    DEVICE = 2;       // uDevice
-    ENTITY = 3;       // uEntity (uE)
-    VERSION = 9;      // uEntity version
-    TOPIC = 4;        // uE Topic
-    METHOD = 5;       // uE Method
-    MESSAGE = 6;      // uE Message
-    RESOURCE = 7;     // uE Resource
-    USER = 8;         // User Information
-  }
+  // Recursively search up to the central level when true, otherwise only search up to the domain level
+  bool search_central = 2;
 }
 
 
-// Message passed to the UpdateNode() RPC call
-message UpdateNodeRequest {
-  // Node to be updated in the database
-  Node node = 1;
+// Response to GetTopicInfoRequest
+message GetServiceTopicsResponse {
+  repeated ServiceTopicInfo topics = 1;
 
-  // Time-to-live: How long (ms) the information should be live for in the database.
-  // -1 means lives forever
-  optional int32 ttl = 2;
+  // How long the topic metadata is valid for in seconds.
+  // If the metadata is older than this value, the client should re-fetch the metadata.
+  // If the field is missed or the value is 0, the client should assume the metadata is valid forever.
+  optional uint32 ttl = 3;
```

Comment:
ServiceTopicInfo already includes ttl, why this one is needed?

Reply:
Correct, I will remove it now